api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +118,59 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestEnvironmentDeleteDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	envTag := models.Tag{
+		EntityID:   "env_id",
+		EntityType: "environment",
+		Key:        "name",
+		Value:      "env_name",
+	}
+
+	if err := tagStore.Insert(envTag); err != nil {
+		t.Fatal(err)
+	}
+
+	var dependencies = []struct {
+		entityID   string
+		entityType string
+		err        string // expected result
+	}{
+		{"lb_id", "load_balancer", error.Error(errors.Newf(errors.DependencyError, "Cannot delete environment 'env_id' because it contains dependent load balancers: lb_id"))},
```

Comment:
How can reduce this part?

Reply:
Take a step back and find the absolute **smallest possible amount** of things need to change between test cases while still testing what you need to. Then, build your table tests around a slice or map of that type (this is usually an [anonymous struct](https://talks.golang.org/2012/10things.slide#3) when there are multiple things that need to change).

Without giving you the answer, I can tell you that we can run these table tests over a slice of strings, e.g.
```
cases := []string{"...", "..."}
for _, c := range cases {
    ...
}
```